packages/langium/src/grammar/type-system/types-util.ts

```diff
@@ -194,3 +194,18 @@ export function findReferenceTypes(type: PropertyType): string[] {
     }
     return [];
 }
+
+export function hasAstType(type: PropertyType): boolean {
+    if (isPropertyUnion(type)) {
+        return type.types.some(hasAstType);
```

Comment:
Should we better use `every`? A union that combines AST types with primitives is not usable as AST type.